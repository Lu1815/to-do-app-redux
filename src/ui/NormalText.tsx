import clsx from 'clsx'

interface INormalTextProps {
    content?: string,
    color?: string,
    weigth?: string
}

const NormalText = ({ content, color, weigth }: INormalTextProps) => {
  return (
    <p className={
        clsx(
            (color ?? 'text-black'),
            (weigth ?? 'font-normal')
        )
    }>
        { content ?? 'Normal Text' }
    </p>
  )
}

export default NormalText