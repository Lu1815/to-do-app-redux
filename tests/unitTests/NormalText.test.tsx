import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NormalText from '../../src/ui/NormalText';

describe('NormalText Component', () => {
  test('renders NormalText with default props', () => {
    render(<NormalText />);
    const textElement = screen.getByText('Normal Text');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveClass('text-black');
    expect(textElement).toHaveClass('font-normal');
  });

  test('renders NormalText with specified content', () => {
    render(<NormalText content="Custom Content" />);
    expect(screen.getByText('Custom Content')).toBeInTheDocument();
  });

  test('renders NormalText with specified color and weight', () => {
    render(<NormalText color="text-red-500" weigth="font-bold" />);
    const textElement = screen.getByText('Normal Text');
    expect(textElement).toHaveClass('text-red-500');
    expect(textElement).toHaveClass('font-bold');
  });

  test('matches snapshot', () => {
    const { container } = render(<NormalText content="Snapshot Test" color="text-blue-500" weigth="font-semibold" />);
    expect(container).toMatchSnapshot();
  });
});
