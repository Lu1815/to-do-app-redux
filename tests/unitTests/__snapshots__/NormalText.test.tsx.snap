// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NormalText Component matches snapshot 1`] = `
<div>
  <p
    class="text-blue-500 font-semibold"
  >
    Snapshot Test
  </p>
</div>
`;
